### ISSUE TITLE
Properly handle connection/channel errors

### ISSUE THREAD

**Issue by jacobcabantomski-ct** [Surnet/graphql-amqp-subscriptions#16]
Per https://github.com/Surnet/graphql-amqp-subscriptions/pull/13/files#r422921589 , we should handle connection/channel errors by attempting to re-establish them when possible. https://github.com/guidesmiths/rascal is a good example to go off of.

**Comment by chdanielmueller** [comment]
Connection errors should not be handled.
I think this is something the user of this library should handle.
We could bubble up an error but everything else seems wrong since it could cause errors if we change something on the shared connection.

Channel error handling an reestablishing those could be worth it.
Maybe even only use one instead of two channels. (Publish and Subscribe use one each)
But the combining of those channels should be discussed and well researched.

Throwing another library in that I have been recommended to use in a scenario like this:
https://www.npmjs.com/package/amqp-connection-manager
For sure needs evaluation which one to use.

**Comment by jacobcabantomski-ct** [comment]
@chdanielmueller Good call, forgot the connection was passed in. This would only be for channel errors. Title and description updated.

**Comment by Kyu0** [comment]
should I resolve this issue?
I tried the amqp-connection-manager library roughly and it seems to be effective. but there are many changes.

**Comment by Kyu0** [comment]
@chdanielmueller Can I work on this?
I tried using the amqp-connection-manager package, and it seems to work.
If you're okay with it, I’ll go ahead and open a PR.

**Comment by chdanielmueller** [comment]
Hi @Kyu0 

The library should only handle channel reconnects and not connection reconnects.
The connection is being passed in and we do not want to change what happens to a connection while using this library.
If you could provide a solution for this feel free to open a PR.
